### ISSUE TITLE
Resurrect "export notebook" feature

### ISSUE THREAD

**Issue by wlach** [iodide-project/iodide#2178]
So we temporarily disabled the "export notebook" feature because the way it worked didn't fit into the server-based paradigm that we moved to. But there are still valid use cases for this!  Maybe someone wants to publish a notebook outside of the context of iodide.io (e.g. #2174), maybe people want to export their data/work so they're not reliant on us. Probably other things I haven't considered.

I would propose we reintroduce this feature, with a few differences from before (at least to start):

1. Exported notebooks will not persist changes. We should probably include a banner in the editor saying that they should re-open this file on an iodide server to do that.
2. Similarly, history, adding files, and other features depending on the server will be disabled.

To start with, I would suggest that the server provide a zipped bundle with the following upon clicking on an export/link button:

* An html file with the iodide notebook inside
* Sufficient javascript/css to render the notebook (can probably use some variant of the webpack stuff we already generate, just need some custom logic for "standalone" mode)
* Any files attached to the notebook

From there, a user can copy them onto a webserver or whatever else they want to do. A subsequent enhancement might be to publish a notebook to github pages.

This wouldn't be a "trivial" feature to write, but probably isn't *too* difficult either. It might  be an interesting feature for a motivated contributor who is comfortable working in both the webpack/js/react and python/django/server worlds. Previous iodide experience highly recommended.

**Comment by bcolloran** [comment]
i think this is blocked by discussion about a few things -- user stories we want to serve (and that we may choose not to serve), potential architectural changes (don't want to paint ourselves into a corner and break the feature again, nor have it slow us down while working through other things), and overall prioritization (i still argue that this falls lower on the list of things that we should be building than a number of other things we've discussed, and that we should also focus on serving the underlying needs/user stories _within_ iodide rather than relying on export functionality as an escape hatch.

anyway, let's have that conversation before anyone begins work on this issue.

thanks for filing this to get the ball rolling @wlach!

**Comment by Jsalaz1989** [comment]
I suspect that I simply don't know enough about Iodide to know exactly how much development effort this feature entails, perhaps I am asking for more than I think, but to clarify my use case (issue #[2174](https://github.com/iodide-project/iodide/issues/2174)), I would simply need a 'snapshot' of the fully evaluated report, not the entire notebook (if that makes sense). So, for example, yes I would need all the HTML and JS necessary to get a pyodide plot to show up with all its buttons and allow for interactions (panning, zooming, home, etc.). But I wouldn't need the notebook itself with run buttons, evaluable cells, etc. If I can embed an interactive plot into my site (an iframe I guess?), I would be happy. If I could just embed the report view, that would be enough. I assume that fulfilling my request is not trivial, but I did want to make sure people didn't think I was asking for more than what I have in mind.

**Comment by wlach** [comment]
Yup, I think exporting a snapshot of the notebook, at least initially, is the way to go. We can discuss this more in the meeting -- I feel like having ways for users to take their content outside iodide is super-important for building trust in the overall ecosystem. and this is an important first step towards that. Anyway, we can certainly discuss this in a meeting when you return.

**Comment by teonbrooks** [comment]
I think this would be quite welcomed by my team. we've converged on using Experimenter and mozilla.report as the sources for our knowledge base. having a snapshot of the report when it has been shared is what we aim to have and it would be on par with our move to provide similar capabilities with [SABER](https://github.com/benmiroglio/saber) using RMarkdown

**Comment by Jsalaz1989** [comment]
Any news on this front?
